.gitlab-ci.yml

```diff
@@ -0,0 +1,186 @@
+      - include/
+      - lib/
+    when:
+      always
+    expire_in: 1 day
+  after_script:
+    - echo "Download debug artifacts from https://gitlab.com/${CI_PROJECT_PATH}/-/jobs"
+
+.script-template: &script-template
+  except:
+    - schedules
+  script:
+    # Create symbolic links under $GOPATH, this is needed for local build
+    - export src=$GOPATH/src
+    - mkdir -p $src/github.com/keroserene
+    - mkdir -p $src/gitlab.com/$CI_PROJECT_NAMESPACE
+    - ln -s $CI_PROJECT_DIR $src/github.com/keroserene/go-webrtc
+    - ln -s $CI_PROJECT_DIR $src/gitlab.com/$CI_PROJECT_PATH
+
+    # build it for go
+    - cd $src/github.com/keroserene/go-webrtc
+    - go get -v .
+    - go build -v -x .
+    - go vet -v .
+    - go test -v -race .
+
+
+# -- jobs ------------------------------------------------------------
+
+debian-stretch_go-1.10:
+  image: golang:1.10-stretch
+  <<: *golang-docker-debian-template
+  <<: *artifacts-template
+  <<: *script-template
+
+debian-stretch_go-1.11:
+  image: golang:1.11-stretch
+  <<: *golang-docker-debian-template
+  <<: *artifacts-template
+  <<: *script-template
+
+ubuntu-devel:
+  image: ubuntu:devel
+  <<: *ubuntu-template
+  <<: *artifacts-template
+  <<: *script-template
+
+ubuntu-rolling:
+  image: ubuntu:rolling
+  <<: *ubuntu-template
+  <<: *artifacts-template
+  <<: *script-template
+
+ubuntu-lts:
+  image: ubuntu:latest
+  <<: *ubuntu-template
+  <<: *artifacts-template
+  <<: *script-template
+
+libwebrtc-linux-amd64-magic:
+  only:
+    - schedules
+  image: debian:stretch
+  # this job needs 20+ gigs of disk
+  tags:
+    - largedisk
+  <<: *artifacts-template
+  script:
+    - *build_env_setup
+    - ./build.sh linux amd64
+
+libwebrtc-linux-arm-magic:
+  only:
+    - schedules
+  image: debian:stretch
+  # this job needs 20+ gigs of disk
+  tags:
+    - largedisk
+  <<: *artifacts-template
+  script:
+    - *build_env_setup
+    - apt-get -qy install --no-install-recommends binutils-arm-linux-gnueabihf
+    - ./build.sh linux arm
+
+android:
+  image: golang:1.11-stretch
+  # this job needs 20+ gigs of disk
+  tags:
+    - largedisk
+  only:
+    - schedules
+  <<: *golang-docker-debian-template
+  <<: *artifacts-template
+  script:
+    - *build_env_setup
+    - apt-get -qy install --no-install-recommends lsb-release sudo
+    - export WEBRTC_SRC="$CI_PROJECT_DIR/third_party/webrtc/src"
+    # crazy workaround for the forced interactive license prompt for Play Services
+    - ./build.sh android arm || true
+    - cat third_party/webrtc/.gclient || true
+    - cat third_party/webrtc/.gclient_entries || true
+    - export ANDROID_HOME=$WEBRTC_SRC/third_party/android_tools/sdk
+    - ls -l $WEBRTC_SRC/third_party || true # DELETEME
+    - printf 'y\ny\ny\ny\ny\ny\ny\ny\n' | $ANDROID_HOME/tools/bin/sdkmanager --licenses
+    - echo y | $ANDROID_HOME/tools/bin/sdkmanager "extras;google;google_play_services"
+    - echo y | $ANDROID_HOME/tools/bin/sdkmanager "extras;google;m2repository"
```

Comment:
Could these tree commands be done uniformly with `yes |` ?

Reply:
this is the standard way to do that, used widely in CI builds for Android.  I've never seen nor tried `yes |`